scripts/split_dataset.py

```diff
@@ -1,91 +1,85 @@
-        elif c["supercategory"] == "light":
-            light_cats.append(c)
-            light_ids.append(c["id"])
-        else:
-            other_cats.append(c)
-            other_ids.append(c["id"])
-
-    for a in data["annotations"]:
-        if a["category_id"] in sign_ids:
-            sign_anns.append(a)
-            sign_img_ids.append(a["image_id"])
-        elif a["category_id"] in light_ids:
-            light_anns.append(a)
-            light_img_ids.append(a["image_id"])
-        else:
-            other_anns.append(a)
-            other_img_ids.append(a["image_id"])
-
-    for i in data["images"]:
-        if i["id"] in sign_img_ids:
-            sign_imgs.append(i)
-        elif i["id"] in light_img_ids:
-            light_imgs.append(i)
-        else:
-            other_imgs.append(i)
-
-    sign_json = {}
-    sign_json["info"] = data["info"]
-    sign_json["licenses"] = data["licenses"]
-    sign_json["categories"] = sign_cats
-    sign_json["images"] = sign_imgs
-    sign_json["annotations"] = sign_anns
-    j = json.dumps(sign_json, indent=4)
-    f = open(file_name + "_signs.json", "w+")
-    f.write(j)
-
-    light_json = {}
-    light_json["info"] = data["info"]
-    light_json["licenses"] = data["licenses"]
-    light_json["categories"] = light_cats
-    light_json["images"] = light_imgs
-    light_json["annotations"] = light_anns
-    j = json.dumps(light_json, indent=4)
-    f = open(file_name + "_lights.json", "w+")
-    f.write(j)
-
-    other_json = {}
-    other_json["info"] = data["info"]
-    other_json["licenses"] = data["licenses"]
-    other_json["categories"] = other_cats
-    other_json["images"] = other_imgs
-    other_json["annotations"] = other_anns
-    j = json.dumps(other_json, indent=4)
-    f = open(file_name + "_other.json", "w+")
-    f.write(j)
+
+    data_path = args.data
+    labels_path = args.labels
+    (label_file_name, _) = os.path.splitext(args.labels)
+    dataset_type = fo.types.COCODetectionDataset
+
+    signs_dataset = fo.Dataset.from_dir( \
+            dataset_type=dataset_type, \
+            data_path=data_path, \
+            labels_path=labels_path, \
+            classes=SIGN_CLASSES, \
+            name="signs_dataset")
+
+    lights_dataset = fo.Dataset.from_dir( \
+            dataset_type=dataset_type, \
+            data_path=data_path, \
+            labels_path=labels_path, \
+            classes=LIGHT_CLASSES, \
+            name="lights_dataset")
+
+    if signs_dataset.stats()["samples_count"] > 0:
+        signs_dataset.export(
+                labels_path=label_file_name + "_signs.json", \
+                dataset_type=dataset_type, \
+                classes=SIGN_CLASSES)
+
+        # Reformat JSON with line breaks and indentation
+
+        f = open(label_file_name + "_signs.json")
+        data = json.load(f)
+        f.close()
+        j = json.dumps(data, indent=4)
+        f = open(label_file_name + "_signs.json", "w+")
+        f.write(j)
+
```

Comment:
Could you please change these `open()` and `close()` blocks to `with open()`. I think the file i/o scope would be a lot more clear, especially since the variable `f` is being reused later. The second `f.open()` is also missing a `f.close()`. Also, you can open the file in both read and write mode, and combine the `json.load()` and `json.dump()` under one scope?

```
with open(label_file_name + "_signs.json", "w+") as f:
    data = json.load(f)
    j = json.dumps(data, indent=4)
    f.write(j)
```

Reply:
Done